diff --git a/Dockerfile b/Dockerfile
index 27c528d..3f61cf1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,6 @@
-ARG FROM_IMAGE_NAME=nvcr.io/nvidia/pytorch:22.04-py3
+ARG FROM_IMAGE_NAME=nvcr.io/nvidia/pytorch:23.10-py3
 FROM ${FROM_IMAGE_NAME}
 
-# Copy code
 WORKDIR /workspace/llm
 COPY . .
 RUN pip install -r requirements.txt
diff --git a/gpt3_blend.sh b/gpt3_blend.sh
index 01312f6..9ec9fea 100755
--- a/gpt3_blend.sh
+++ b/gpt3_blend.sh
@@ -1,8 +1,10 @@
 #!/bin/bash
 
-COM_DIR="/c4/preprocessed_c4_googlespm"
+COM_DIR="${COM_DIR:?COM_DIR not set}"
+
 C4_0="${COM_DIR}/c4_en_6_c4_spm_text_document"
 C4_1="${COM_DIR}/c4_en_7_c4_spm_text_document"
 DATA_BLEND="0.5 ${C4_0} 0.5 ${C4_1}"
+
 VALID_C4="${COM_DIR}/c4_en_validation_subset_c4_spm_text_document"
 VALID_DATA_BLEND="1.00 ${VALID_C4}"
diff --git a/megatron/optimizer/clip_grads.py b/megatron/optimizer/clip_grads.py
index ad249bd..22fe827 100755
--- a/megatron/optimizer/clip_grads.py
+++ b/megatron/optimizer/clip_grads.py
@@ -16,7 +16,7 @@
 """Gradient clipping."""
 
 import torch
-from torch._six import inf
+from torch import inf
 
 from apex.multi_tensor_apply import multi_tensor_applier
 import amp_C
diff --git a/run_gpt3.sh b/run_gpt3.sh
index ce35f2b..12384cc 100755
--- a/run_gpt3.sh
+++ b/run_gpt3.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-#SBATCH -p luna -A mlperf -t 00:20:00 --nodes=8 --exclusive --mem=0 --overcommit --ntasks-per-node=8 --job-name=mlperf-megatron:megatron
+#SBATCH -A mlperf -t 00:20:00 --nodes=8 --exclusive --mem=0 --overcommit --ntasks-per-node=8 --job-name=mlperf-megatron:megatron
 
 # Vars without defaults
 LOG_DIR=${1:?LOG_DIR not set}
@@ -107,10 +107,13 @@ fi
 run_cmd="python -u ${MEGATRON_DIR}/pretrain_gpt.py ${options}"
 DATETIME=`date +'date_%y-%m-%d_time_%H-%M-%S'`
 
+export NCCL_TOPO_FILE='/var/run/nvidia-topologyd/virtualTopology.xml'
+
 srun -l \
      --container-image $CONT \
-     --container-mounts "$PWD:$PWD,${COM_DIR}:${COM_DIR},${LOG_DIR}:${LOG_DIR},${BPE_DIR}:${BPE_DIR}" \
+     --container-mounts "$NCCL_TOPO_FILE:$NCCL_TOPO_FILE,$PWD:$PWD,${COM_DIR}:${COM_DIR},${LOG_DIR}:${LOG_DIR},${BPE_DIR}:${BPE_DIR}" \
+     --container-writable \
+     --export NCCL_TOPO_FILE \
      --output=$LOG_DIR/GPT3-175B-runlog-$DATETIME.log sh -c "${run_cmd}"
 
 set +x
-
